content/docs/tech-insights/track-sonarqube/index.md

```diff
@@ -0,0 +1,121 @@
+
+1. First identify the project key for the catalog Component.
+You can find this in the Sonarqube dashboard. Find the project, click Administration -> Update key and copy the "Project Key"
+2. Next edit the catalog YAML file for the Component 
+The easiest way to do this is to find the component in the Roadie catalog and click the edit pencil in the top right hand side of the About Card.
+3. Edit and save the YAML file to include the `sonarqube.org/project-key` annotation.
+e.g.
+```yaml
+---
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: test-react-app
+  annotations:
+    sonarqube.org/project-key: punkle_test-react-app
+```
+
+## How to configure and run the Roadie Sonarqube broker application to connect Sonarqube with Roadie
+
+1. First generate a new token in the Sonarqube dashboard
+Click on your profile in the top right of the application, click My Account. Then click Security. Enter a token name and click "Generate Token". Copy the token value.
+2. Run the Roadie broker client application to connect Sonarqube with backstage.
+```shell
+docker run \
+  --env SONARQUBE_INSTANCE_URL=https://<sonarqube_address> \
+  --env BROKER_TOKEN=sonarqube \
+  --env BROKER_SERVER_URL=https://<tenant_name>.broker.roadie.so \
+  --env SONARQUBE_TOKEN=<sonarqube_token> \
+  roadiehq/broker:5691884253.43.1-sonarqube
+```
+3. Determine the IP address from where the broker is running and set this IP in the broker allow list.
+To set the IP allow list go to Administration > Settings > Broker enter the IP address and subnet mask using CIDR notation.
+4. Confirm that the broker client by looking at the docker container logs. You should see the following text:
+```text
+successfully established a websocket connection to the broker server
+```
+
+## Create an HTTP Data Source to collect data from Sonarqube
+
+1. Within Roadie, click on Tech Insights -> Data Sources and click "Add Datasource"
+2. Set the name to "Sonarqube", and set the description if you prefer.
+3. Select the HTTP Data Provider, then select the proxy we created earlier.
+4. In the Path Extension enter `sonarqube/api/measures/component?component={{metadata.annotations["sonarqube.org/project-key"]}}&metricKeys=bugs,security_rating,duplicated_lines_density,security_hotspots`
+5. Next select an entity that you know has been configured with the Sonarcloud annotation so that we can test the data extraction in the next step.
+6. Click view. You should see the result of a successful HTTP response from Sonarqube.
+7. Next you can select the fields to extract from the API.
+In the Field extraction section select the "JSON with JSONata syntax" extractor. Then enter "bugs" as the fact name, `$.component.measures[metric = "bugs"].value` as the JSONata query, and select "Integer" type. Add a second fact by clicking "Add Fact", enter "security_hotspots", `$.component.measures[metric = "security_hotspots"].value` and select "Integer" again.
```

Comment:
Using a table is a nice way to make this more scannable. See [this example](https://roadie.io/docs/tech-insights/enforce-branch-protection/).

Reply:
done.